cpp/src/parquet/arrow/reader.cc

```diff
@@ -682,6 +709,18 @@ Status OpenFile(const std::shared_ptr<::arrow::io::ReadableFileInterface>& file,
                   reader);
 }
 
+Status OpenFile(const std::shared_ptr<::arrow::io::ReadableFileInterface>& file,
```

Comment:
This is deprecated I think, should be `::arrow::io::RandomAccessFile`. I opened https://issues.apache.org/jira/browse/ARROW-4789